examples/picamera/pitrack.py

```diff
@@ -0,0 +1,182 @@
+        num_larger_packets = payload_size % num_packets
+        package_size = payload_size // num_packets
+
+        f_nri = data[0] & (0x80 | 0x60)  # fni of original header
+        nal = data[0] & 0x1F
+
+        fu_indicator = f_nri | NAL_TYPE_FU_A
+
+        fu_header_end = bytes([fu_indicator, nal | 0x40])
+        fu_header_middle = bytes([fu_indicator, nal])
+        fu_header_start = bytes([fu_indicator, nal | 0x80])
+        fu_header = fu_header_start
+
+        packages = []
+        offset = NAL_HEADER_SIZE
+        while offset < len(data):
+            if num_larger_packets > 0:
+                num_larger_packets -= 1
+                payload = data[offset : offset + package_size + 1]
+                offset += package_size + 1
+            else:
+                payload = data[offset : offset + package_size]
+                offset += package_size
+
+            if offset == len(data):
+                fu_header = fu_header_end
+
+            packages.append(fu_header + payload)
+
+            fu_header = fu_header_middle
+        assert offset == len(data), "incorrect fragment data"
+
+        return packages
+
+    @staticmethod
+    def _packetize_stap_a(
+        data: bytes, packages_iterator: Iterator[bytes]
+    ) -> Tuple[bytes, bytes]:
+        counter = 0
+        available_size = PACKET_MAX - STAP_A_HEADER_SIZE
+
+        stap_header = NAL_TYPE_STAP_A | (data[0] & 0xE0)
+
+        payload = bytes()
+        try:
+            nalu = data  # with header
+            while len(nalu) <= available_size and counter < 9:
+                stap_header |= nalu[0] & 0x80
+
+                nri = nalu[0] & 0x60
+                if stap_header & 0x60 < nri:
+                    stap_header = stap_header & 0x9F | nri
+
+                available_size -= LENGTH_FIELD_SIZE + len(nalu)
+                counter += 1
+                payload += pack("!H", len(nalu)) + nalu
+                nalu = next(packages_iterator)
+
+            if counter == 0:
+                nalu = next(packages_iterator)
+        except StopIteration:
+            nalu = None
+
+        if counter <= 1:
+            return data, nalu
+        else:
+            return bytes([stap_header]) + payload, nalu
+
+    @classmethod
+    def _packetize(cls, packages: Iterator[bytes]) -> List[bytes]:
+        packetized_packages = []
+
+        packages_iterator = iter(packages)
+        package = next(packages_iterator, None)
+        while package is not None:
+            if len(package) > PACKET_MAX:
+                packetized_packages.extend(cls._packetize_fu_a(package))
+                package = next(packages_iterator, None)
+            else:
+                packetized, package = cls._packetize_stap_a(package, packages_iterator)
+                packetized_packages.append(packetized)
+
+        return packetized_packages
+
+    @staticmethod
+    def _split_bitstream(buf: bytes) -> Iterator[bytes]:
```

Comment:
Is this the same as the implementation in h264.py? Any way to unify that code?

Reply:
Its the same but different  but keeping outside of the library allow the user to define it own way to split the bitstream for its own format.